### PR TITLE
Requests

### PR DESCRIPTION
**Pull request by RazumauMikita**
# What type of PR is this? (check all applicable)
- [x] 🍕 Feature
# Description
1. Login form is finalized.
2. /auth, /users requests ate created.
3. Register form is created.
4. Link to register page within login page has been created.

### CODE REVIEW

File: pet-finder/src/Components/Forms/LoginForm/index.tsx [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 import { LoginFormType, loginSchema } from '../../../utils/loginFormSchema'
+import { ERROR_MESSAGES } from '../../../constants/errorMessages'
+import { IResponseBodyLogIn } from '../../../requests/interfaces'
+import {
+  exceptionResponse,
+  ExceptionMessage,
+} from '../../../requests/constants'
+import { loginFormFields } from '../../../constants/formFields'
 
 import styles from './LoginForm.module.scss'
```
Comment (Andrew-Nes) [inline pet-finder/src/Components/Forms/LoginForm/index.tsx:19]: empty interface LoginFormProps

File: pet-finder/src/Components/Forms/RegistrationForm/index.tsx [hunk 2]
```diff
@@ -1,9 +1,90 @@
...
+} from '../../../requests/interfaces'
+import {
+  ExceptionMessage,
+  exceptionResponse,
+} from '../../../requests/constants'
 
 import styles from './RegistrationForm.module.scss'
+import { registerFormFields } from '../../../constants/formFields'
 
 interface RegistrationFormProps {}
```
Comment (Andrew-Nes) [inline pet-finder/src/Components/Forms/RegistrationForm/index.tsx]: empty interface (yagni)

File: pet-finder/src/Components/Forms/LoginForm/index.tsx [hunk 3]
```diff
@@ -16,28 +24,48 @@ export const LoginForm: FC<LoginFormProps> = () => {
...
-  }
+    setError,
+  } = useForm({
+    mode: 'all',
+    resolver: yupResolver(loginSchema(ERROR_MESSAGES)),
+  })
+
+  const onSubmit: SubmitHandler<LoginFormType> = useCallback(
+    async ({ email, password }) => {
+      console.log(email, password)
```
Comment (Andrew-Nes) [inline pet-finder/src/Components/Forms/LoginForm/index.tsx]: no console

File: pet-finder/src/Components/Forms/LoginForm/index.tsx [hunk 4]
```diff
@@ -16,28 +24,48 @@ export const LoginForm: FC<LoginFormProps> = () => {
...
+      try {
+        const response = await logIn({ email, password })
+        if (!response.ok) {
+          const serverMessage =
+            exceptionResponse[response.status as keyof ExceptionMessage]
+          setError('root.serverError', { message: serverMessage })
+        } else {
+          const responseBody: IResponseBodyLogIn = await response.json()
+          console.log(responseBody)
```
Comment (Andrew-Nes) [inline pet-finder/src/Components/Forms/LoginForm/index.tsx:41]: same, console log

File: pet-finder/src/Components/Forms/RegistrationForm/index.tsx [hunk 5]
```diff
@@ -1,9 +1,90 @@
...
+        } else {
+          const reqBodyLogIn: IRequestBodyLogIn = {
+            email: requestBody.email,
+            password: requestBody.password,
+          }
+          const logInResponse = await logIn(reqBodyLogIn)
+          if (logInResponse.ok) {
+            const { accessToken, refreshToken }: IResponseBodyLogIn =
+              await logInResponse.json()
+            console.log(accessToken, refreshToken)
```
Comment (Andrew-Nes) [inline pet-finder/src/Components/Forms/RegistrationForm/index.tsx:61]: console

File: pet-finder/src/requests/req.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      Authorization: `Bearer ${accessToken}`,
+    },
+  })
+
+  const responseBody: ResponseBodyUserById = await response.json()
+  return responseBody
+}
+// export const postLost = async () => {
+
+// }
```
Comment (Andrew-Nes) [inline pet-finder/src/requests/req.ts]: commented code section

File: pet-finder/src/Components/Forms/LoginForm/index.tsx (hunk 3 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/LoginForm/index.tsx:35]: Lets leave this logs, work is not finished here

File: pet-finder/src/Components/Forms/LoginForm/index.tsx (hunk 4 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/LoginForm/index.tsx:44]: Lets leave this logs, work is not finished here

File: pet-finder/src/Components/Forms/LoginForm/index.tsx (hunk 1 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/LoginForm/index.tsx:19]: 4c2728b066835430aa3ca0889233c9a452184030
This interface has been removed

File: pet-finder/src/requests/req.ts (hunk 6 above)
Comment (RazumauMikita) [reply pet-finder/src/requests/req.ts]: 4c2728b066835430aa3ca0889233c9a452184030
Commented section has been removed

File: pet-finder/src/Components/Forms/RegistrationForm/index.tsx (hunk 2 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/RegistrationForm/index.tsx]: 4c2728b066835430aa3ca0889233c9a452184030
Empty interface has been removed

File: pet-finder/src/Components/Forms/LoginForm/index.tsx (hunk 4 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/LoginForm/index.tsx:41]: Lets leave this log, work in not finished here

File: pet-finder/src/Components/Forms/RegistrationForm/index.tsx (hunk 5 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/RegistrationForm/index.tsx:61]: Lets leave this log, work in not finished here

File: pet-finder/src/Components/Forms/LoginForm/index.tsx (hunk 3 above)
Comment (RazumauMikita) [reply pet-finder/src/Components/Forms/LoginForm/index.tsx]: 321e12ec4ec3718dbefae19e7d049a0da7d1c8b2
I Removed this log

**Review by Andrew-Nes — APPROVED** [review]

**Pull request merged**